src/pybind/mgr/orchestrator.py

```diff
@@ -0,0 +1,456 @@
+    result read of e.g. device inventory than it is to keep the caller waiting
+    while you scan hosts every time.
+    """
+
+    def is_orchestrator_module(self):
+        """
+        Enable other modules to interrogate this module to discover
+        whether it's usable as an orchestrator module.
+
+        Subclasses do not need to override this.
+        """
+        return True
+
+    def available(self):
+        """
+        Report whether we can talk to the orchestrator.  This is the
+        place to give the user a meaningful message if the orchestrator
+        isn't running or can't be contacted.
+
+        This method may be called frequently (e.g. every page load
+        to conditionally display a warning banner), so make sure it's
+        not too expensive.  It's okay to give a slightly stale status
+        (e.g. based on a periodic background ping of the orchestrator)
+        if that's necessary to make this method fast.
+
+        Do not override this method if you don't have a meaningful
+        status to return: the default None, None return value is used
+        to indicate that a module is unable to indicate its availability.
+
+        @return two-tuple of boolean, string
+        """
+        return None, None
+
+    def wait(self, completions):
+        """
+        Given a list of Completion instances, progress any which are
+        incomplete.  Return a true if everything is done.
+
+        Callers should inspect the detail of each completion to identify
+        partial completion/progress information, and present that information
+        to the user.
+
+        For fast operations (e.g. reading from a database), implementations
+        may choose to do blocking IO in this call.
+        """
+        raise NotImplementedError()
+
+    def get_inventory(self, node_filter=None):
+        """
+
+        :param node_filter:
+        :return: list of InventoryNode
+        """
+        raise NotImplementedError()
+
+    def describe_service(self, service_type, service_id):
+        """
+        Describe a service (of any kind) that is already configured in
+        the orchestrator.  For example, when viewing an OSD in the dashboard
+        we might like to also display information about the orchestrator's
+        view of the service (like the kubernetes pod ID).
+
+        When viewing a CephFS filesystem in the dashboard, we would use this
+        to display the pods being currently run for MDS daemons.
+        """
+        raise NotImplementedError()
+
+    def create_osds(self, osd_spec):
+        """
+        Create one or more OSDs within a single Drive Group.
+
+        The principal argument here is the drive_group member
+        of OsdSpec: other fields are advisory/extensible for any
+        finer-grained OSD feature enablement (choice of backing store,
+        compression/encryption, etc).
+
+        :param osd_spec: OsdCreationSpec
+        :return:
```

Comment:
this is empty, and renders oddly, if there is nothing to describe here maybe remove it